apps/extension/src/routes/page/onboarding/height.tsx

```diff
@@ -0,0 +1,73 @@
+import { BackIcon } from '@repo/ui/components/ui/icons/back-icon';
+import { Button } from '@repo/ui/components/ui/button';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from '@repo/ui/components/ui/card';
+import { FadeTransition } from '@repo/ui/components/ui/fade-transition';
+import { usePageNav } from '../../../utils/navigate';
+import { PagePath } from '../paths';
+import { FormEvent, MouseEvent, useState } from 'react';
+import { Input } from '@repo/ui/components/ui/input';
+import { localExtStorage } from '../../../storage/local';
+
+export const ImportWalletCreationHeight = () => {
+  const navigate = usePageNav();
+  const [blockHeight, setBlockHeight] = useState('');
+
+  const handleSubmit = async (event: MouseEvent | FormEvent) => {
+    event.preventDefault();
+
+    if (blockHeight) {
+      // Save the block height to local extension storage
+      await localExtStorage.set('walletCreationBlockHeight', Number(blockHeight));
+    }
```

Comment:
question: Local storage setting seems to happen at the very end of onboarding right? See `useOnboardingSave()` hook. If you use zustand, you can just add logic there to save `walletCreationBlockHeight`

Reply:
opted to implement this, which required explicitly differentiating between fresh and existing wallets, as they require _different_ zustand slice interfaces.

Reply:
removed zustand slices